packages/neos-ui-editors/src/Editors/Image/index.js

```diff
@@ -215,8 +215,16 @@ export default class ImageEditor extends Component {
     }
 
     handleChooseFile = () => {
-        this.previewScreen.chooseFromLocalFileSystem();
-        this.setState({isAssetLoading: true});
+        const {secondaryEditorsRegistry, options} = this.props;
+        if (secondaryEditorsRegistry.get('Neos.Neos/Inspector/Secondary/Editors/AssetUploadScreen')) {
+            // set media type constraint to "image/*" if it is not explicitly specified via options.constraints.mediaTypes
+            const constraints = {...options.constraints, mediaTypes: (options.constraints && options.constraints.mediaTypes) || ['image/*']};
+            const {component: AssetUploadScreen} = secondaryEditorsRegistry.get('Neos.Neos/Inspector/Secondary/Editors/AssetUploadScreen');
+            this.props.renderSecondaryInspector('IMAGE_UPLOAD_MEDIA', () => <AssetUploadScreen type="images" constraints={constraints} onComplete={this.afterUpload}/>);
```

Comment:
Is setting the `isAssetLoading` state not also relevant in this branch?

Reply:
I don't know if showing the spinner is relavant, if a secondary editor with an upload dialog is shown. Addtional there is a problem when closing the secondary editor via the close button in the top right corner. If I set the `isAssetLoading` to true, I don't see a good way to reset the state to false when the secondary editor is closed via the close button. I think, a possible way could be an optional third parameter for the function `renderSecondaryInspector` for adding a `handleCloseSecondaryEditor` function. This function could than be used the adjust the behaviour of `onClose` by rewriting the `handleCloseSecondaryInspector` funtions in the inspector.